### PR TITLE
Milestone api

### PR DESCRIPTION
**Pull request by leszko11**
Hey there 👋 

Added api methods for [milestones](https://docs.github.com/en/rest/issues/milestones?apiVersion=2022-11-28#list-milestones)

### CODE REVIEW

**Comment by pietbrauer** [comment]
@leszko11 Please rebase from `main` where I fixed the CI.

**Comment by leszko11** [comment]
Ah, finally 😅 
Sorry for this commits spam

File: OctoKit/Milestone.swift [hunk 1]
```diff
@@ -67,3 +68,397 @@ open class Milestone: Codable {
...
+
+// MARK: Request
+
+public extension Octokit {
+    /**
+     Create a milestone
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repo: The name of the repository.
+     - parameter title: The title of the new milestone.
+     - parameter state: Filter pulls by their state.
```
Comment (pietbrauer) [inline OctoKit/Milestone.swift:80]: This seems to be incorrect. Creating a milestone should have no filter?

File: OctoKit/Milestone.swift [hunk 2]
```diff
@@ -67,3 +68,397 @@ open class Milestone: Codable {
...
+    }
+
+    #if compiler(>=5.5.2) && canImport(_Concurrency)
+    @available(macOS 12.0, iOS 15.0, tvOS 15.0, watchOS 8.0, *)
+    /**
+     Create a milestone
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repo: The name of the repository.
+     - parameter title: The title of the new milestone.
+     - parameter state: Filter pulls by their state.
```
Comment (pietbrauer) [inline OctoKit/Milestone.swift:114]: Same here?

File: OctoKit/Milestone.swift [hunk 3]
```diff
@@ -67,3 +68,397 @@ open class Milestone: Codable {
...
+        decoder.dateDecodingStrategy = .formatted(Time.rfc3339DateFormatter)
+        return try await router.post(session, decoder: decoder, expectedResultType: Milestone.self)
+    }
+    #endif
+
+    /**
+     Get a list of milestones
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repo: The name of the repository.
+     - parameter state: Filter pulls by their state.
```
Comment (pietbrauer) [inline OctoKit/Milestone.swift:181]: Same here?

File: OctoKit/Milestone.swift [hunk 4]
```diff
@@ -67,3 +68,397 @@ open class Milestone: Codable {
...
+            }
+        }
+    }
+
+    #if compiler(>=5.5.2) && canImport(_Concurrency)
+    /**
+     Get a list of milestones
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repo: The name of the repository.
+     - parameter state: Filter pulls by their state.
```
Comment (pietbrauer) [inline OctoKit/Milestone.swift:215]: Same here?

File: OctoKit/Milestone.swift [hunk 5]
```diff
@@ -67,3 +68,397 @@ open class Milestone: Codable {
...
+    }
+    #endif
+
+    /**
+     Update a milestone
+     - parameter owner: The user or organization that owns the repositories.
+     - parameter repo: The name of the repository.
+     - parameter number: The number that identifies the milestone.
+     - parameter title: The title of the new milestone.
+     - parameter state: Filter pulls by their state.
```
Comment (pietbrauer) [inline OctoKit/Milestone.swift:242]: Same here?

File: OctoKit/Milestone.swift (hunk 1 above)
Comment (leszko11) [reply OctoKit/Milestone.swift:80]: The docs are wrong, but create milestone endpoint accepts `state` value
https://docs.github.com/en/rest/issues/milestones?apiVersion=2022-11-28#create-a-milestone

Comment (leszko11) [inline OctoKit/Milestone.swift:80]:
```suggestion
     - parameter state: The state of the milestone
```

**Review by pietbrauer — APPROVED** [review]
Thanks 🚀

**Pull request merged**